packages/wrangler/src/assets.ts

```diff
@@ -538,18 +519,65 @@ export function validateAssetsArgsAndConfig(
 
 	// Smart placement turned on when using assets
 	if (
-		config?.placement?.mode === "smart" &&
-		config?.assets?.run_worker_first === true
+		config.placement?.mode === "smart" &&
+		config.assets?.run_worker_first === true
 	) {
 		logger.warn(
 			"Turning on Smart Placement in a Worker that is using assets and run_worker_first set to true means that your entire Worker could be moved to run closer to your data source, and all requests will go to that Worker before serving assets.\n" +
 				"This could result in poor performance as round trip times could increase when serving assets.\n\n" +
 				"Read more: https://developers.cloudflare.com/workers/static-assets/binding/#smart-placement"
 		);
 	}
+
+	if (
+		(args.assets || config.assets?.directory) &&
+		!(args.script || config.main)
+	) {
+		const unsupportedConfigKeys = getConfigKeysUnsupportedByAssetsOnly(config);
+
+		if (unsupportedConfigKeys.length > 0) {
+			const keys = unsupportedConfigKeys.map((key) => `⋅ "${key}"`).join("\n");
+
+			throw new UserError(
```

Comment:
are we abs sure we want to throw instead of warn?

Reply:
We lock down these fields in the dash. EWC does accept them, but it also would accept `{ "complete": "junk" }` keys (we're not "strict").

I'd say error sets people up for the most success, but I leave the final decision to ANT. If you'd prefer warning, go for it.

Reply:
For this case, I think a warning would be better -- we can warn that these fields will be locked down in the Dashboard (assuming this means non-editable). @penalosa what do you think?